webapp/web-components/cht-form/README.md

```diff
@@ -0,0 +1,103 @@
+# cht-form Web Component
+
+This [Angular Element](https://angular.io/guide/elements) encapsulates the functionality required for displaying and interacting with an Enketo form into a custom [Web Component](https://developer.mozilla.org/en-US/docs/Web/Web_Components). The `cht-form` web component can be used in any web page to display and interact with an Enketo form. The form layout and behavior will be functionally identical to form interactions within the CHT webapp.
+
+## Building the web component
+
+To build the web component, run the following commands from the root directory of cht-core:
+
+```shell
+npm ci
+npm run build-cht-form
+```
+
+This will build the `cht-form` component and place the build artifacts in the `./build/cht-form` directory.
+
+## Using the web component
+
+The `cht-form` web component can be included on any page by pulling in the following files:
+
+- `runtime.js`
+- `polyfills.js`
+- `scripts.js`
+- `main.js`
+- `styles.css`
+
+With those files included on the page, you can then use the `cht-form` web component in your HTML:
+
+```html
+<cht-form id="myform"></cht-form>
+```
+
+### Form inputs
+
+The following inputs are supported by the `cht-form` web component:
+
+- `formHtml` (Required) - String value containing the HTML content of the form to render.
+- `formModel` (Required) - String value containing the model xml of the form to render.
+- `formXml` (Required) - String value containing the ODK xform xml of the form to render.
+- `contactType` (Required for contact forms) - String value containing the contact type of the form to render (e.g. `person`, `district_hospital`, etc).
+- `user` - The user's settings document from the CouchDB `_users` database.
+- `contactSummary` - The context data for the contact summary of the form's contact. Must not be set for `contact`
+  forms.
+- `content` - The content data to provide when rendering the form. The `contact` field can be the hydrated contact
+  document for the form's contact.
+
+These inputs can be set directly onto the `cht-form` HTML element:
+
+```js
+const myForm = document.getElementById('myform');
+myForm.user = { contact_id: 'test_user' };
+myForm.contactSummary = { pregnancy_uuid: 'myPregUUID' };
+myForm.content = { contact: { name: "My Test Patient" } };
+
+myForm.formHtml = formData.formHtml;
+myForm.formModel = formData.formModel;
+myForm.formXml = formData.formXml;
+```
+
+### Form outputs
+
+Outputs from the web component are emitted as events. The following events are emitted:
+
+- `onRender` - Emitted when the form is finished rendering.
+- `onSubmit` - Emitted when the form is submitted (the submit button is pressed). The event detail contains an array of
+  docs created by the form.
+- `onCancel` - Emitted when the form is cancelled (the cancel button is pressed).
+
+```js
+myForm.addEventListener('onSubmit', async (e) => {
+  console.log('form submitted', e.detail);
+});
+```
```

Comment:
The event name feels a bit unnatural when typing it, what do you think about dropping the `on` prefix?  
```js
form.addEventListener('submit', () => {});
```
or would it conflict with other event names?

Reply:
So the `onSubmit` event is one of the original events from the existing Enketo component: hhttps://github.com/medic/cht-core/blob/ef28068b5c5076c7b52b64912f6af8f230e68f2e/webapp/src/ts/components/enketo/enketo.component.ts#L17

I don't think there is any reason that we _have_ to keep the same naming, but I did try and maintain as much consistency as possible between the Enekto component and the new cht-form web component (one day we might try to use cht-form to replace the Enekto component).  Alternatively, I could try to update the event naming both in the web component and in the Enekto component.  

Let me know if you think we should pursue any of these alternatives in this PR, or just leave it as it is for now. (Admitedly the risk of changing this later is that we would break any consumer of the Web Component that relies on these events (e.g. the test-harness).

Reply:
Right, I hadn't realized those were the same event names from our existing enketo component. I would say let's stay consistent with our APIs and keep it this way.

Worst case scenario, if we do want to change it later down the line, we can emit to both `onSubmit` and `submit` before deprecating `onSubmit` with a major version. So, some breaking changes but definitely manageable.